### PR TITLE
Restore the permissions for public keys

### PR DESCRIPTION
**Pull request by Acidity**
These were removed in #249 to fix performance issues

### CODE REVIEW

File: grouper/api/handlers.py [hunk 1]
```diff
@@ -110,6 +110,8 @@ def get(self, name=None):
                 details = self.graph.get_user_details(name, cutoff)
             else:
                 return self.notfound("User (%s) not found." % name)
+            for key in md["public_keys"]:
+                key["permissions"] = key["permissions"]()
```
Comment (lfaraone) [inline grouper/api/handlers.py]: uh, wha? What is this doing? :)

Comment (lfaraone) [inline grouper/api/handlers.py]: Oh. Got it. I feel like this would be less surprising as a generator?

File: grouper/api/handlers.py [hunk 2]
```diff
@@ -110,6 +111,7 @@ def get(self, name=None):
                 details = self.graph.get_user_details(name, cutoff)
             else:
                 return self.notfound("User (%s) not found." % name)
+            md = copy.deepcopy(md)
```
Comment (lfaraone) [inline grouper/api/handlers.py]: is this germane to this commit or the previous one?

File: grouper/graph.py [hunk 3]
```diff
@@ -179,9 +179,10 @@ def user_indexify(data):
                         "created_on": str(key.created_on),
                         "tags": [tag.name for tag in public_key_tags.get(key.id, [])],
                         "permissions": lambda: list(set([(perm.name, perm.argument) for perm in
-                                                   get_public_key_permissions(session,
-                                                        PublicKey.get(session, id=key.id)
-                                                    )
+                                                   get_public_key_permissions(
+                                                       session,
+                                                       PublicKey.get(session, id=key.id),
+                                                   )
```
Comment (lfaraone) [inline grouper/graph.py]: changes to unrelated files for syntax purity are best separated out. 

`git add -p` ftw

File: tests/test_audit.py [hunk 4]
```diff
@@ -93,6 +93,8 @@ def test_audit_end_to_end(session, users, groups, http_client, base_url):  # noq
     add_member(groups[groupname], users["zay@a.co"])
     add_member(groups[groupname], users["gary@a.co"])
 
+    graph.update_from_db(session)
+
```
Comment (lfaraone) [inline tests/test_audit.py]: Is this change needed?

Comment (Acidity) [inline tests/test_audit.py]: Yes. Prior to this change user_permissions did not rely on the graph being up to date so the test didn't need to enforce the graph being updated in order to work properly. Now however, user_permissions does need the graph to be up to date in order to provide correct information, so we need to force it to be up to date here.
